Source/WebCore/page/LocalFrame.h

```diff
@@ -404,6 +415,10 @@ class LocalFrame final : public Frame {
     SandboxFlags m_sandboxFlags;
     UniqueRef<EventHandler> m_eventHandler;
     HashSet<RegistrableDomain> m_storageAccessExceptionDomains;
+
+#if ENABLE(CONTENT_EXTENSIONS)
+    RefPtr<ResourceMonitor> m_resourceMonitor;
```

Comment:
Do frames ever share ResourceMonitors? Shouldn't same-origin iframes share the same monitor? Otherwise, what's preventing a frame from creating many same-origin subframes to do a lot of networking within them?

Reply:
I think about that design but are there any chance the frame is moved into different frame? In that case, I cannot think of easy way to recalculate the usage. Or even we need to recalculate the usage or not. Then it's safe to make the monitor for each iframe.

Reply:
Can we store on the Document then? Documents definitely do not get moved. Frames can change URL (be navigated) and I don't think we want to track resources usage across navigations anyway?

Reply:
That sounds great. I try that.

Reply:
Let me clarify your suggestion. You suggested that the `ResourceMonitor` should be shared among same origin frames and to make the network usage won't be affected when iframe is moved into different parent, store the usage in `document` instead of in `ResourceMonitor` directly. Is this correct?

Reply:
No, I'm asking if the `ResourceMonitor` should be on Document instead of `Frame`. A Frame gets re-used on navigation, a Document doesn't.

Reply:
> A Frame gets re-used on navigation

Oh, didn't know that. Got it.

Reply:
I may misunderstand the initial comment. I thought you've asked that two same origin frames should share one `ResourceMonitor`. Is this something you've concerned or not?

Reply:
Yes, I do think that Documents sharing the same origin should share the same ResourceMonitor or at least the same quota. Otherwise, it is too easy to work around limits by creating new same-origin iframes. It doesn't have to be in this first PR but we'll have to deal with it before enabling the feature.